packages/pwa-kit-runtime/src/utils/ssr-config.server.js

```diff
@@ -28,21 +30,26 @@ const SUPPORTED_FILE_TYPES = ['js', 'yml', 'yaml', 'json']
  * Each file marked with `ext` can optionally be terminated with `js`, `yml|yaml` or
  * `json`. The file loaded is also determined based on that precidence of file extension.
  *
+ * @param {Object} opts.buildDirectory - Option path to the folder containing the configution. Byt default
+ * it is the `build` folder when running remotely and the project folder when developing locally.
  * @returns - the application configuration object.
  */
 /* istanbul ignore next */
-export const getConfig = () => {
-    const isRemote = Object.prototype.hasOwnProperty.call(process.env, 'AWS_LAMBDA_FUNCTION_NAME')
+export const getConfig = (opts = {}) => {
+    const {buildDirectory} = opts
+    const configDirBase = IS_REMOTE ? 'build' : ''
     let targetName = process?.env?.DEPLOY_TARGET || ''
 
     const targetSearchPlaces = SUPPORTED_FILE_TYPES.map((ext) => `config/${targetName}.${ext}`)
     const localeSearchPlaces = SUPPORTED_FILE_TYPES.map((ext) => `config/local.${ext}`)
     const defaultSearchPlaces = SUPPORTED_FILE_TYPES.map((ext) => `config/default.${ext}`)
 
+    const searchFrom = buildDirectory || process.cwd() + '/' + configDirBase
```

Comment:
🤔 what's the value for `buildDirectory` here? Do we need to prepend it with `process.cwd()` perhaps?

Reply:
The the idea here was to have the `buildDirectory` option in the 'build' command be symmetrical with the `buildDirectory` option in the 'push' commend. So that being said, it can accept both a relative path and an absolute path. 

The weird thing I just spotted after you brought this up was that `cosmiconfig` didn't really care if the buildDirectory existed or not, because by default it will start traversing the file tree upwards looking for configuration files. 

This behaviour in the context of pushing wasn't great because it would take a while to fail since it's possible that it found a package.json file further up the file tree and used that for the push config, then only failed during the build step. 

So I took extra precaution and  decided to throw if the buildDirectory you supply doesn't exist. 

So after all that, to answer you question we are taking the absolute or relative path and passing it to cosmiconfig, which from the documentation doesn't specify if it has to be relative or absolute. It's possible that it resolves it internally.